frame/staking/src/lib.rs

```diff
@@ -1627,23 +1719,38 @@ decl_module! {
 
 		/// Force there to be a new era at the end of sessions indefinitely.
 		///
+		/// The dispatch origin must be Root.
+		///
 		/// # <weight>
-		/// - One storage write
+		/// - Base Weight: 2.429 µs
+		/// - Write: ForceEra
 		/// # </weight>
-		#[weight = 5_000_000]
+		#[weight = 2 * WEIGHT_PER_MICROS + T::DbWeight::get().writes(1)]
 		fn force_new_era_always(origin) {
 			ensure_root(origin)?;
 			ForceEra::put(Forcing::ForceAlways);
 		}
 
-		/// Cancel enactment of a deferred slash. Can be called by either the root origin or
-		/// the `T::SlashCancelOrigin`.
-		/// passing the era and indices of the slashes for that era to kill.
+		/// Cancel enactment of a deferred slash.
+		///
+		/// Can be called by either the root origin or the `T::SlashCancelOrigin`.
+		///
+		/// Parameters: era and indices of the slashes for that era to kill.
 		///
 		/// # <weight>
-		/// - One storage write.
+		/// - Base: 5880 + 34.62 * S µs
+		/// - Read: Unapplied Slashes
+		/// - Write: Unapplied Slashes
 		/// # </weight>
-		#[weight = 1_000_000_000]
+		#[weight = FunctionOf(
+			|(_, slash_indices,): (&EraIndex, &Vec<u32>,)| {
+				T::DbWeight::get().reads_writes(1, 1)
+					.saturating_add(5_880 * WEIGHT_PER_MICROS)
```

Comment:
This weight was determined using always 1,000 pending slashes, and then removing some variable amount of them.

Reply:
Why is it `5870` above in the docs?

Reply:
this was a mistake, should be 5870